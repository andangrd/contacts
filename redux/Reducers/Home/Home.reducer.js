import { createAction } from 'redux-actions';
import typeToReducer from 'type-to-reducer';
// import { getSuggestion } from '../../../services/Home/Home.service'


// ------------------------------------
// Constants
// ------------------------------------
const SET_WELCOME_MESSAGE = 'HOME/SET_WELCOME_MESSAGE';
const SET_QUERY = 'HOME/SET_QUERY';
const GET_AUTOCOMPLETE_DATA = 'HOME/GET_AUTOCOMPLETE_DATA';

export const actionTypes = {
    SET_WELCOME_MESSAGE,
    SET_QUERY,
    GET_AUTOCOMPLETE_DATA
};

// ------------------------------------
// Action
// ------------------------------------
export const setWelcomeMessage = createAction(SET_WELCOME_MESSAGE);
export const setQuery = createAction(SET_QUERY, (query) => Promise.resolve(query));
// export const getAutocompleteData = createAction(GET_AUTOCOMPLETE_DATA, getSuggestion);

export const actions = {
    setWelcomeMessage,
    setQuery,
    // getAutocompleteData
};


// ------------------------------------
// Reducer
// ------------------------------------

// initial state
export const initialState = {
    welcomeMessage: 'Open up the code for this screen, bro',
    query: '',
    autoCompleteData: []
};

const setWelcomeMessageHandler = (state, action) => ({
    ...state,
    welcomeMessage: action.payload
});

const setQueryHandler = (state, action) => ({
    ...state,
    query: action.payload
});

const getAutocompleteDataHandler = (state, action) => ({
    ...state,
    autoCompleteData: action.payload
});

const getAutoComleteError = (state) => ({
    ...state,
    autoCompleteData: []
});

// type to reducers
export default typeToReducer({
    // [GET_AUTOCOMPLETE_DATA]: {
    //     FULFILLED: getAutocompleteDataHandler,
    //     REJECTED: getAutoComleteError
    // },
    [SET_WELCOME_MESSAGE]: setWelcomeMessageHandler,
    [SET_QUERY]: {
        FULFILLED: setQueryHandler
    }
}, initialState);