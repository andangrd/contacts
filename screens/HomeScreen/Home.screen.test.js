import React from 'react';
import renderer from 'react-test-renderer';

import { HomeScreen } from './Home.screen';


it(`renders correctly`, () => {
  jest.useFakeTimers()
  const tree = renderer.create(< HomeScreen />).toJSON();

  expect(tree).toMatchSnapshot();
});
