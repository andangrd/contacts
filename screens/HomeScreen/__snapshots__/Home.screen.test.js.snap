// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F4F1DE",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    keyboardShouldPersistTaps="handled"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#F4F1DE",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginVertical": 50,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 70,
            "marginVertical": 30,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "left": 0,
              "position": "relative",
              "right": 0,
              "top": 0,
              "zIndex": 100,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "zIndex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#b9b9b9",
                    "borderRadius": 1,
                    "borderWidth": 1,
                  },
                  Object {
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "backgroundColor": "white",
                    "borderRadius": 20,
                    "flex": 1,
                    "flexDirection": "row",
                    "height": 40,
                    "justifyContent": "center",
                    "paddingLeft": 3,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "backgroundColor": "white",
                    "borderRadius": 20,
                    "flex": 1,
                    "flexDirection": "row",
                    "height": 40,
                    "justifyContent": "center",
                    "paddingLeft": 3,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  collapsable={false}
                  data={Array []}
                  flatListProps={Object {}}
                  hideResults={true}
                  inputContainerStyle={
                    Object {
                      "alignItems": "stretch",
                      "alignSelf": "stretch",
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "flex": 1,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                      "paddingLeft": 3,
                    }
                  }
                  keyExtractor={[Function]}
                  keyboardShouldPersistTaps="always"
                  listContainerStyle={
                    Object {
                      "flex": 1,
                    }
                  }
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onEndEditing={[Function]}
                  onFocus={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onStartShouldSetResponderCapture={[Function]}
                  rejectResponderTermination={true}
                  renderItem={[Function]}
                  renderSeparator={null}
                  renderTextInput={[Function]}
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "justifyContent": "space-around",
                      "width": "80%",
                    }
                  }
                  underlineColorAndroid="transparent"
                />
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                      "opacity": 1,
                      "paddingLeft": 5,
                      "width": "20%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "paddingRight": 3,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
